src/MassTransit.Abstractions/IMetricsContext.cs

```diff
@@ -0,0 +1,15 @@
+﻿namespace MassTransit
+{
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// Provides access to tags added to metrics.
+    /// </summary>
+    public interface IMetricsContext
+    {
+        /// <summary>
+        /// Gets the tag collection.
+        /// </summary>
+        ICollection<KeyValuePair<string, object?>> Tags { get; }
+    }
+}
```

Comment:
You're also going to need to add some extension methods to actually _add_ the tags. Something like.

```cs
public static void AddMetricsTag(this PipeContext context, string name, string value)
{
    context.AddOrUpdatePayload<MetricsContext>(() => new TagList(){name,value}, e => e.Add(name,value));
}
```
}